packages/backend-storage/API.md

```diff
@@ -23,9 +22,13 @@ export type AmplifyStorageFactoryProps = Omit<AmplifyStorageProps, 'outputStorag
 
 // @public (undocumented)
 export type AmplifyStorageProps = {
+    isDefault?: boolean;
     name: string;
     versioned?: boolean;
-    outputStorageStrategy?: BackendOutputStorageStrategy<StorageOutput>;
+    outputStorageStrategy?: BackendOutputStorageStrategy<{
+        version: '1';
+        payload: Record<string, string>;
+    }>;

```

Comment:
why this change?
Shouldn't we change `StorageOutput` type ?

Reply:
StorageOutput is from schema, which is the final output, but the initial output we stored in stack metadata is different, so I introduced a new type, `StorageOutputToStore` in a new commit [f6a1c12](https://github.com/aws-amplify/amplify-backend/pull/1742/commits/f6a1c121a2238c021c9c09c29281a18257090af7)

Reply:
See how the new commit [f6a1c12](https://github.com/aws-amplify/amplify-backend/pull/1742/commits/f6a1c121a2238c021c9c09c29281a18257090af7) looks?

Reply:
I don't understand why do we need type `StorageOutputToStore` .

Auth and Functions are just using their respective outputs, see:
https://github.com/aws-amplify/amplify-backend/blob/7361ff4ed9da10f9b48e327c96eff25eb81a9e10/packages/auth-construct/src/construct.ts#L990
and
https://github.com/aws-amplify/amplify-backend/blob/7361ff4ed9da10f9b48e327c96eff25eb81a9e10/packages/backend-function/src/factory.ts#L238

Reply:
> StorageOutput is from schema

`StorageOutput` is a shape that we store in CFN metadata/outputs. This is not related to client config schemas - that translation happens when client config package reads stacks and transforms it to amplify outputs.

Reply:
There would be some mis-match after `CfnOutput` sending the output ([code](https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-output-storage/src/stack_metadata_output_storage_strategy.ts#L86)), then the [output](https://github.com/aws-amplify/amplify-backend/blob/main/packages/deployed-backend-client/src/backend_output_client.ts#L24) we get from CFN would be like the following

```
{
 version: "1",
 payload: {
   name: [..., ..., ...],
   bucketName: [..., ..., ...],
   storageRegion: [..., ..., ...],
 }
}
```

So we need to change the type, and modify the output when parsing to `amplify_outputs.json` ([code](https://github.com/aws-amplify/amplify-backend/blob/main/packages/deployed-backend-client/src/backend_output_client.ts#L25C45-L25C51)).

Is there a way to avoid the change?

Reply:
tl;dr -- When we store output, we store the individual bucket ([code](https://github.com/aws-amplify/amplify-backend/blob/main/packages/backend-storage/src/construct.ts#L153)) (which has `name`, `bucketName`, and `storageRegion`), but when generate the amplify_outputs.json, we also need to add `buckets`, and remove `name` etc. So, there are two different `storageOutput` . 

There might be a way to avoid the complex, and I missed it.

Reply:
The way we should evolve this is
Change
```
export const storageOutputSchema = z.object({
  version: z.literal('1'),
  payload: z.object({
    bucketName: z.string(),
    storageRegion: z.string(),
  }),
});
```

To

```
export const storageOutputSchema = z.object({
  version: z.literal('1'),
  payload: z.object({
    bucketName: z.string(),
    storageRegion: z.string(),
    buckets: z.string(), // JSON array as string (serialized buckets, like in functions output), this should likely be optional.
  }),
});

```

And then in order for this to be backwards compatible we should populate bucketName and storageRegion with the one that's 'isDefault = true'.

Reply:
And then use this trick https://github.com/aws-amplify/amplify-backend/blob/890ce865cd9988d504878ea115104e01cfdfbc47/packages/backend-function/src/factory.ts#L393-L398 to append to `buckets`.

Reply:
Done d67854ec809441ddd20bc88407be944aa6950d42